### ISSUE TITLE
Document ordering / sort algorithm

### ISSUE THREAD

**Issue by jorhett** [voxpupuli/puppet-rsyslog#45]
## Affected Puppet, Ruby, OS and module versions/distributions

- Puppet: 5.3
- Ruby: 2.4
- Distribution: ?
- Module version: 

## How to reproduce (e.g Puppet code you use)

```
mail:
    priority: 110
    type: 'omfile'
    facility: 'mail.*'
    config:
      file: '/var/log/maillog'
  cron:
    priority: 111
    type: 'omfile'
    facility: 'cron.*'
    config:
      file: '/var/log/cron.log'
```

## What are you seeing

These rules are places in the rsyslog file after global definitions (priority 10) but before templates, config, etc (20, 30...) To get it to sort properly I have had to use the following:

```
mail:
    priority: 6510
  cron:
    priority: 6511
```

## What behaviour did you expect instead

Priority is shown in the examples as an integer. One assumes it would be sorted like an integer.  I actually kindof like the effect of being able to just make longer numbers to sort within the section-- but the documentation makes no mention of this!

**Comment by dhollinger** [comment]
@jorhett The documentation makes no mention of sort ordering because it's simply using the `order` attribute from `puppetlabs-concat` where such functionality is already documented. Perhaps that should be noted within the README (Which I have no problem with).

However, if the ordering isn't working as expected that could be an issue related to `puppetlabs-concat` rather than `puppet-rsyslog`. 

I will attempt to reproduce your issue and see what I can come up with.

**Comment by dhollinger** [comment]
@jorhett Looks like this is tied to how [puppetlabs-concat](https://github.com/puppetlabs/puppetlabs-concat) either functions or is a bug in that module. After some testing I get the same results when using the `concat` and `concat::fragment` defined types by themselves.

I will update the documentation to note what I've found and will close this issue when that has been fixed. However, it does appear that the ordering sees any number after the first 2 digits as some sort of "sub-ordering" or something similar
